packages/opentelemetry-instrumentation-replicate/opentelemetry/instrumentation/replicate/__init__.py

```diff
@@ -0,0 +1,165 @@
+"""OpenTelemetry Replicate instrumentation"""
+import logging
+import os
+from typing import Collection
+from wrapt import wrap_function_wrapper
+
+from opentelemetry import context as context_api
+from opentelemetry.trace import get_tracer, SpanKind
+from opentelemetry.trace.status import Status, StatusCode
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    _SUPPRESS_INSTRUMENTATION_KEY,
+    unwrap,
+)
+
+from opentelemetry.semconv.ai import SpanAttributes, LLMRequestTypeValues
+from opentelemetry.instrumentation.replicate.version import __version__
+
+logger = logging.getLogger(__name__)
+
+_instruments = ("replicate >= 0.22.0",)
+
+WRAPPED_METHODS = [
+    {
+        "module": "replicate",
+        "method": "run",
+        "span_name": "replicate.run",
+    },
+    {
+        "module": "replicate",
+        "method": "stream",
+        "span_name": "replicate.stream",
+    },
+    {
+        "module": "replicate",
+        "method": "predictions.create",
+        "span_name": "replicate.predictions.create",
+    },
+]
+
+
+def should_send_prompts():
+    return (
+        os.getenv("TRACELOOP_TRACE_CONTENT") or "true"
+    ).lower() == "true" or context_api.get_value("override_enable_content_tracing")
+
+
+def _set_span_attribute(span, name, value):
+    if value is not None:
+        if value != "":
+            span.set_attribute(name, value)
+    return
+
+
+def _set_input_attributes(span, kwargs):
```

Comment:
This depends on the exact model we use. What I'd want to see here is some "dynamic" way of extracting the arguments and sending them as span attributes, with things like temperature indeed in the semantic convention names if they exist. This is an example for Llama:
https://replicate.com/meta/llama-2-70b-chat/api?tab=python